tensorboard/plugins/pr_curve/tf_pr_curve_dashboard/tf-pr-curve-card.html

```diff
@@ -0,0 +1,306 @@
+        },
+        _tooltipColumns: {
+          type: Array,
+          readOnly: true,
+          value: () => {
+            const valueFormatter = ChartHelpers.multiscaleFormatter(
+                ChartHelpers.Y_TOOLTIP_FORMATTER_PRECISION);
+            const formatValueOrNaN =
+                (x) => isNaN(x) ? 'NaN' : valueFormatter(x);
+            return [
+              {
+                title: 'Name',
+                computeFunction: (d) => d.dataset.metadata().name,
+              },
+              {
+                title: 'Precision',
+                computeFunction: (d) => formatValueOrNaN(d.datum.precision),
+              },
+              {
+                title: 'Recall',
+                computeFunction: (d) => formatValueOrNaN(d.datum.recall),
+              }];
+          },
+        }
+      },
+      observers: [
+          'reload(run, tag)',
+          '_runsChanged(_attached, runs.*)',
+          '_setChartData(_runToPrCurveEntry)',
+      ],
+      _computeRunColor(run) {
+        return this._colorScaleFunction.scale(run);
+      },
+      attached() {
+        // Defer reloading until after we're attached, because that ensures that
+        // the requestManager has been set from above. (Polymer is tricky
+        // sometimes)
+        this._attached = true;
+        this.reload();
+      },
+      reload() {
+        if (!this._attached) {
+          return;
+        }
+        this._canceller.cancelAll();
+        const router = getRouter();
+        let url = router.pluginRoute('pr_curves', '/pr_curves');
+        url += url.indexOf('?') > -1 ? '&' : '?';
+        const urlRunsPortion = this.runs.map(r => `&run=${r}`).join('');
+        url += `tag=${this.tag}${urlRunsPortion}`;       
+        const updateData = this._canceller.cancellable(result => {
+          if (result.cancelled) {
+            return;
+          }
+
+          const runToData = result.value;
+          this.set('_runToData', runToData);
+        });
+        this.requestManager.request(url).then(updateData);
+      },
+      _setChartData(actualPrCurveEntryPerRun) {
+        _.forOwn(actualPrCurveEntryPerRun, (entry, run) => {
+          // Reverse the values so they are plotted in order. The logic within
+          // the line chart for associating information to show in the tooltip
+          // with points in the chart assumes that the series data is ordered
+          // by the variable on the X axis. If the values are not in order,
+          // tooltips will not work because the tooltip will always be stuck on
+          // one side of the chart. 
+          const precision = entry.precision.slice().reverse();
+          const recall = entry.recall.slice().reverse();
+          const seriesData = _.zipWith(
+              precision, recall, (precision, recall) => ({
+                precision: precision,
+                recall: recall,
+              }));
+          this.$$('vz-line-chart').setSeriesData(run, seriesData);
```

Comment:
It would be great to find a way to not have to rerender _all_ series when just one changes.

Perhaps using a wildcard observer (`_runToPrCurveEntry.*`) will work. I could imagine it failing, though, if a run name contains a literal period, so you'll have to be able to test that. You'll also want to carefully test what happens when you select and deselect runs as active.

Otherwise, storing a `_previousActualPrCurveEntryPerRun` or something might be the best way to go. Even if you have to compare all the values, it'll surely still be faster than the re-rendering computations.